app/services/dollar_parser.rb

```diff
@@ -1,5 +1,16 @@
-class DollarParser < BaseParser
-  def separator
-    '$'
+class DollarParser
```

Comment:
Instead of inheritance use composition.
There might some duplication but I feel the parser are decoupled enough for future changes to them.